libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -430,23 +430,34 @@ size_t AP_Scheduler::task_info(char *buf, size_t bufsize)
     bufsize -= n;
     total += n;
 
+    // baseline the total time taken by all tasks
+    float total_time = 1.0f;
+    for (uint8_t i = 0; i < _num_tasks; i++) {
+        const AP::PerfInfo::TaskInfo* ti = perf_info.get_task_info(i);
+        if (ti->tick_count > 0) {
+            total_time += ti->elapsed_time_us;
+        }
+    }
+
```

Comment:
it may be interesting to calculate 100.0f / total_time instead of in the loop. I haven't check the compiler output but it would make sens to prevent to do multiple time the division